tests/src/Kernel/ViewsLinkFieldsTest.php

```diff
@@ -0,0 +1,149 @@
+        parent::setUp();
+
+        $this->installEntitySchema('file');
+        $this->installEntitySchema('media');
+        $this->installEntitySchema('user');
+        $this->installConfig(['user', 'views', 'collabora_online_test']);
+        $this->installSchema('file', ['file_usage']);
+        // Install user module avoid user 1 permissions bypass.
+        \Drupal::moduleHandler()->loadInclude('user', 'install');
+        user_install();
+
+        // Create two medias to check access with different scopes, 'any' and 'own'.
+        $this->createMediaEntity('document');
+        $this->ownMedia = $this->createMediaEntity('document');;
+    }
+
+    /**
+     * Tests link fields.
+     */
+    public function testLinks(): void {
+        // User without permissions can't see links.
+        $this->doTestLinks([
+            'preview' => [FALSE, FALSE],
+            'edit' => [FALSE, FALSE],
+        ],
+        $this->createUser([]));
+        // User with 'Preview' permission can see preview link.
+        $this->doTestLinks([
+            'preview' => [TRUE, TRUE],
+            'edit' => [FALSE, FALSE],
+        ],
+        $this->createUser([
+        'preview document in collabora'
+        ]));
+        // User with 'Edit any' permission can see edit link.
+        $this->doTestLinks([
+            'preview' => [FALSE, FALSE],
+            'edit' => [TRUE, TRUE],
+        ],
+        $this->createUser([
+            'edit any document in collabora'
+        ]));
+        // User with 'Edit own' permission can see edit link for entities they own.
+        $this->doTestLinks([
+            'preview' => [FALSE, FALSE],
+            'edit' => [FALSE, TRUE],
+        ],
+    $this->createUser([
+            'edit own document in collabora'
+        ]));
+    }
+
+    /**
+     * Tests that links behave as expected.
+     *
+     * @param [] $expected_results
+     *   An associative array of expected results keyed by operation.
+     * @param \Drupal\Core\Session\AccountInterface $account
+     *   The user account to be used to run the test;
+     */
+    protected function doTestLinks(array $expected_results, AccountInterface $account) {
+        $this->setCurrentUser(account: $account);
+        // Set the current user as the owner to check 'edit own' access.
+        $this->ownMedia->setOwnerId($account->id())->save();
+        $view = Views::getView('test_collabora_links');
+        $view->preview();
+
+        $info = [
+        'preview' => [
+            'label' => 'Preview in Collabora',
+            'field_id' => 'collabora_preview',
+            'route' => 'collabora-online.view'
+        ],
+        'edit' => [
+            'label' => 'Edit in Collabora',
+            'field_id' => 'collabora_edit',
+            'route' => 'collabora-online.edit'
+        ],
+        ];
+
+        $i = 0;
+        // Check each expected results for every media.
+        foreach(Media::loadMultiple() as $media) {
```

Comment:
I see now we are using the `$i` to identify which of the two media entities we are looking at.
I think this can be made more clear like this:
(of course then we have to drop the `$i++;`.
```suggestion
        $media_entities = Media::loadMultiple();
        // Check each expected results for every media.
        foreach(array_values($media_entities) as $i => $media) {
```
or
```suggestion
        $media_entities = Media::loadMultiple();
        // Check each expected results for every media.
        foreach(array_keys($media_entities) as $i => $media_id) {
```
We could also rename the `$i` to `$i_media`. But I think it is not needed anymore, if the `$i` is the foreach key.

Reply:
Yes, that would be better but loadMultiple uses entity IDs, so can't be used as keys.

Reply:
I'm thinking and the $i can be overlooked, Moving the comment avobe it will help to keep the reader attetion to it. About the index change, its just another way of dealing with indexes, the simplier the better, so I keep it as is.